immich-app/immich
# chore(ml): added testing and github workflow

## Description

This PR adds `pytest` unit tests to the ML package. Given the expanding scope of ML, it's important to ensure a certain set of behaviors across changes. The tests included here aren't exhaustive and should be built upon in the future, but they're a good starting point.

Further, formatting, linting and testing are all automatically checked through GHA to ensure consistency across contributions.

## Testing

To test locally, you can follow these steps:
1. `cd machine-learning`
2. `poetry install --with dev`
3. `pytest --cov`

To test in a container, you must copy or map the `machine-learning` folder, not just `app`. You can then SSH into the container and run `pytest --cov`.

Note that models are mocked in most tests to avoid downloading and loading them. The exceptions are the endpoint tests, which are skipped by default. Setting `MACHINE_LEARNING_TEST_FULL` will run these tests as well.
